R/shortTermChange.R

```diff
@@ -0,0 +1,189 @@
+#' tse <- tse[, !(colData(tse)$Time_label %in% short_time_labels)]
+#' tse <- tse[, (colData(tse)$StudyIdentifier == "Bioreactor A")]
+#' 
+#' # Plot short term change in abundance
+#' shortTermChange(tse, rarefy = TRUE, plot = TRUE) + ggtitle("Bioreactor A")
+
+
+#' @rdname shortTermChange
+#' @export
+setGeneric("shortTermChange", signature = c("x"),
+    function(x,assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), ...)
+        standardGeneric("shortTermChange"))
+
+#' @rdname shortTermChange
+#' @export
+#' @importFrom dplyr arrange as_tibble summarize
+#' @importFrom ggplot2 ggplot aes
+#' @importFrom ggrepel geom_text_repel
+#' @importFrom mia rarefyAssay transformAssay
+#' @importFrom SummarizedExperiment colData
+setMethod("shortTermChange", signature = c(x = "SummarizedExperiment"),
+    function(x, assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), plot = FALSE, ...){
+        ############################## Input check #############################
+        # Check validity of object
+        if(nrow(x) == 0L){
+            stop("No data available in `x` ('x' has nrow(x) == 0L.)",
+                 call. = FALSE)
+        }
+        # Check assay.type
+        mia:::.check_assay_present(assay.type, x)
+        if(!mia:::.is_a_bool(rarefy)){
+            stop("'rarefy' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if(!mia:::.is_a_bool(compositional)){
+            stop("'compositional' must be TRUE or FALSE.", call. = FALSE)
+        }
+        # Ensure that the provided depth is valid
+        if ( !is.null(depth) && depth > min(assay(x, assay.type)) ) {
+            stop("Depth cannot be greater than the minimum number of counts in your data")
+        
+        }
+        ############################ Input check end ###########################
+        
+        ############################ Data Preparation #########################
+        # Initialize the filtered object based on rarefy and compositional arguments
+        if (rarefy == TRUE && compositional == FALSE) {
+            message("rarefy is set to TRUE, calculating short term change using counts")
+            x <- rarefyAssay(x, assay.type = assay.type, depth = depth)
+            assay.type <- "subsampled"
+        } else if (rarefy == FALSE && compositional == FALSE) {
+            message("rarefy is set to FALSE, compositional==FALSE, using raw counts")
+            x <- x
+        } else if (rarefy == FALSE && compositional == TRUE) {
+            message("rarefy is set to FALSE, compositional==TRUE, using relative abundances")
+            x <- transformAssay(x, method = "relabundance", assay.type = assay.type)
+            assay.type <- "relabundance"
+        } else if (rarefy == TRUE && compositional == TRUE) {
+            stop("Both rarefy and compositional cannot be TRUE simultaneously")
+        }
```

Comment:
This should be done outside the function

Reply:
yes would be better so

Reply:
done.